client/src/components/App.js

```diff
@@ -35,6 +43,56 @@ class App extends Component {
     this.setState({projects});
   }
 
+  parseProjectForm(form_data) {
+    for (let key in form_data) {
+      if (form_data[key] === '') {
+        // Actually delete so that we don't send empty strings or undefined to the database.
+        delete form_data[key];
+      }
+    }
+    if (!form_data.title || !form_data.link || !form_data.email) {
+      // TODO: show error here
+      return;
+    }
+    return form_data;
+  }
+
+  async submitNewProject(form_data) {
+    let data = this.parseProjectForm(form_data);
+    const response = await fetch('/projects', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    const result = await response.json();
+    // TODO handle error here
+    this.showProjects(); // TODO change server to send back project in json, then use the result
+    console.log('response,', result);
+    this.hideForm();
+  }
+
+  async submitEditedProject(id, form_data) {
```

Comment:
Wow does this function ever look like the `submitNewProject` function. How about we merge them so that we don't need to add error handling twice? 😉